src/main/java/de/fraunhofer/aisec/cpg/graph/Node.java

```diff
@@ -178,7 +178,7 @@ public void removePrevEOGEntry(@NonNull Node eog) {
   public void removePrevEOGEntries(@NonNull List<Node> prevEOGs) {
     for (Node n : prevEOGs) {
       List<PropertyEdge<Node>> remove =
-          PropertyEdge.findPropertyEdgesByPredicate(this.prevEOG, e -> e.getStart().equals(n));
+          PropertyEdge.findPropertyEdgesByPredicate(this.prevEOG, e -> e.getStart() == n);
```

Comment:
This one itself did not have influence on the behavior of the test or the bugs. I just used it as Nodes may be equal according to the equals implementation but are still two separate nodes. 

The main reason why I had to change the PropertyEdge.equals method was because it did not involve the nodes connected to the edges, which lead to List<PropertyEdge>.removes(Node) to remove all PropertyEdges that have the same properties although one of the connected Nodes is different.

Reply:
[https://github.com/Fraunhofer-AISEC/cpg/blob/f11ed5094e7f8b80925942e1e6c8847c8b05d6db/src/main/java/de/fraunhofer/aisec/cpg/graph/Node.java#L182](List remove in Node.java)